[test optimization] Add test session name `DD_TEST_SESSION_NAME` instructions (#27436)
DataDog/documentation: content/en/tests/setup/java.md

`MAVEN_OPTS=-javaagent:$DD_TRACER_FOLDER/dd-java-agent.jar` (Required)
: Injects the tracer into the Maven build process.

`DD_TEST_SESSION_NAME`

> : Use this to identify a group of tests (for example: `unit-tests` or `integration-tests`).

Reviewer: ```suggestion
: Identifies a group of tests (for example: `unit-tests` or `integration-tests`).
```

`DD_CIVISIBILITY_ENABLED=true` (Required)
: Enables the Test Optimization product.

`DD_TEST_SESSION_NAME`

> : Use this to identify a group of tests (for example: `unit-tests` or `integration-tests`).

Reviewer: ```suggestion
: Identifies a group of tests (for example: `unit-tests` or `integration-tests`).
```

`DD_CIVISIBILITY_ENABLED=true` (Required)
: Enables the Test Optimization product.

`DD_TEST_SESSION_NAME`
: Use this to identify a group of tests (for example: `unit-tests` or `integration-tests`).

> `DD_ENV` (Required)

Reviewer 2: Should we remove `DD_ENV` then, or at least mark it as optional, to be consistent across languages?

Author: I'll remove the `(Required)` part but leave the rest, so the user knows the env var is there

}
```

### Test session name `DD_TEST_SESSION_NAME`

> Use `DD_TEST_SESSION_NAME` to define the test session name for your tests (`test_session.name` tag). Use this to identify a group of tests. Examples of values for this tag would be:

Reviewer: Same as above comment

Author: https://github.com/DataDog/documentation/pull/27436/files#r1946692292

- `unit-tests`
- `integration-tests`
- `smoke-tests`
- `flaky-tests`
- `ui-tests`
- `backend-tests`

> If `DD_TEST_SESSION_NAME` is not specified, the default value used is a combination of:

Reviewer: ```suggestion
If `DD_TEST_SESSION_NAME` is not specified, the default value used is a combination of the:
```

- CI job name
- Command used to run the tests (such as `mvn test`)

> The test session name should be unique within a repository to help you distinguish different groups of tests.

Reviewer: ```suggestion
The test session name needs to be unique within a repository to help you distinguish different groups of tests.
```

#### When to use `DD_TEST_SESSION_NAME`

If your tests are run with commands that include a dynamic string, such as:

- `mvn test --temp-dir=/var/folders/t1/rs2htfh55mz9px2j4prmpg_c0000gq/T`

> The default value for the test session name will be unstable. It is recommended to use `DD_TEST_SESSION_NAME` in this case.

Reviewer: Same question as above comment

> The default value for the test session name will be unstable. It is recommended to use `DD_TEST_SESSION_NAME` in this case.

Reviewer: ```suggestion
The default value for the test session name will be unstable. Datadog recommends using `DD_TEST_SESSION_NAME` in this case.
```

}
```

### Test session name `DD_TEST_SESSION_NAME`

Use `DD_TEST_SESSION_NAME` to define the test session name for your tests (`test_session.name` tag). Use this to identify a group of tests. Examples of values for this tag would be:

- `unit-tests`
- `integration-tests`
- `smoke-tests`
- `flaky-tests`
- `ui-tests`
- `backend-tests`

If `DD_TEST_SESSION_NAME` is not specified, the default value used is a combination of:

- CI job name

> - Command used to run the tests (such as `yarn test`)

Reviewer 2: I think something Java-specific, like `mvn test` or `./gradlew test`, would look more natural here :)

Author: oh sry I wanted to do this but I missed it, thanks
